src/app/main/apps/vehicles/store/vehiclesSlice.js

```diff
@@ -29,103 +28,104 @@ export const getVehicles = createAsyncThunk('vehicle-list-app/vehicles/getVehicl
+    dispatch(getVehicles());
 
-//     return data;
-//   }
-// );
+    return data;
+  }
+);
 
-// export const updateVehicle = createAsyncThunk(
-//   'vehiclesApp/vehicles/updateVehicle',
-//   async (vehicle, { dispatch, getState }) => {
-//     const response = await axios.post('/api/vehicles-app/update-vehicle', { vehicle });
-//     const data = await response.data;
+export const updateVehicle = createAsyncThunk('vehiclesApp/vehicles/updateVehicle', async (vehicle, { dispatch }) => {
+  const response = await axios.put(`${VEHICLES_API}/${vehicle.id}`, vehicle, {
+    headers: {
+      Authorization: TOKEN
+    }
+  });
+  const { data } = response;
 
-//     dispatch(getVehicles());
+  dispatch(getVehicles());
 
-//     return data;
-//   }
-// );
+  return data;
+});
 
-// export const removeVehicle = createAsyncThunk(
-//   'vehiclesApp/vehicles/removeVehicle',
-//   async (vehicleId, { dispatch, getState }) => {
-//     await axios.post('/api/vehicles-app/remove-vehicle', { vehicleId });
+export const removeVehicle = createAsyncThunk(
+  'vehiclesApp/vehicles/removeVehicle',
+  async (vehicleId, { dispatch, getState }) => {
+    await axios.post('/api/vehicles-app/remove-vehicle', { vehicleId });
 
-//     return vehicleId;
-//   }
-// );
+    return vehicleId;
+  }
+);
 
-// export const removeVehicles = createAsyncThunk(
-//   'vehiclesApp/vehicles/removeVehicles',
-//   async (vehicleIds, { dispatch, getState }) => {
-//     await axios.post('/api/vehicles-app/remove-vehicles', { vehicleIds });
+export const removeVehicles = createAsyncThunk(
+  'vehiclesApp/vehicles/removeVehicles',
+  async (vehicleIds, { dispatch, getState }) => {
+    await axios.post('/api/vehicles-app/remove-vehicles', { vehicleIds });
 
-//     return vehicleIds;
-//   }
-// );
+    return vehicleIds;
+  }
+);
 
-// export const toggleStarredVehicle = createAsyncThunk(
-//   'vehiclesApp/vehicles/toggleStarredVehicle',
-//   async (vehicleId, { dispatch, getState }) => {
-//     const response = await axios.post('/api/vehicles-app/toggle-starred-vehicle', { vehicleId });
-//     const data = await response.data;
+export const toggleStarredVehicle = createAsyncThunk(
+  'vehiclesApp/vehicles/toggleStarredVehicle',
+  async (vehicleId, { dispatch, getState }) => {
+    const response = await axios.post('/api/vehicles-app/toggle-starred-vehicle', { vehicleId });
+    const data = await response.data;
 
-//     dispatch(getUserData());
+    dispatch(getUserData());
 
-//     dispatch(getVehicles());
+    dispatch(getVehicles());
 
-//     return data;
-//   }
-// );
+    return data;
+  }
+);
 
-// export const toggleStarredVehicles = createAsyncThunk(
-//   'vehiclesApp/vehicles/toggleStarredVehicles',
-//   async (vehicleIds, { dispatch, getState }) => {
-//     const response = await axios.post('/api/vehicles-app/toggle-starred-vehicles', { vehicleIds });
-//     const data = await response.data;
+export const toggleStarredVehicles = createAsyncThunk(
```

Comment:
we don't have this function in out app - pls comment out this function ?

Reply:
okey i will comment this